sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -645,15 +645,15 @@ def _raw_predict(self, X):
         raw_predictions : array, shape (n_trees_per_iteration, n_samples)
             The raw predicted values.
         """
-        X = check_array(X, dtype=[X_DTYPE, X_BINNED_DTYPE],
-                        force_all_finite=False)
+        is_binned = getattr(self, '_in_fit', False)
+        dtype = [X_BINNED_DTYPE] if is_binned else [X_DTYPE]
```

Comment:
in master, X would still be of dtype `X_BINNED_DTYPE` (pass-through) but `is_binned` would be False so we'd call `_predict_from_numeric_data` which can only accept `X_DTYPE`, hence the failure